app/src/main/java/org/simlar/service/liblinphone/LinphoneHandler.java

```diff
@@ -377,4 +383,150 @@ private static void enableDebugMode(final boolean enabled)
+		mLinphoneCore.updateCall(currentCall, params);
+	}
+
+	public void preventAutoAnswer()
+	{
+		Lg.i("preventAutoAnswer");
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to prevent auto answer for");
+			return;
+		}
+
+		try {
+			mLinphoneCore.deferCallUpdate(currentCall);
+		} catch (final LinphoneCoreException e) {
+			Lg.ex(e, "LinphoneCoreException during deferCallUpdate");
+		}
+	}
+
+	public void acceptVideoUpdate(final boolean accept)
+	{
+		Lg.i("acceptVideoUpdate accept=", accept);
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to accept video for");
+			return;
+		}
+
+		final LinphoneCallParams params = currentCall.getCurrentParams();
+		if (accept) {
+			params.setVideoEnabled(true);
+		}
+
+		try {
+			mLinphoneCore.acceptCallUpdate(currentCall, params);
+		} catch (final LinphoneCoreException e) {
+			Lg.ex(e, "LinphoneCoreException during acceptCallUpdate");
+		}
+	}
+
+	public void setVideoWindow(final Object videoWindow)
+	{
+		Lg.i("setVideoWindow");
+
+		if (mLinphoneCore == null) {
+			Lg.e("setVideoWindow: mLinphoneCore is null => aborting");
+			return;
+		}
+
+		mLinphoneCore.setVideoWindow(videoWindow);
+	}
+
+	public void setVideoPreviewWindow(final Object videoPreviewWindow)
+	{
+		Lg.i("setVideoPreviewWindow");
+
+		if (mLinphoneCore == null) {
+			Lg.e("setVideoPreviewWindow: mLinphoneCore is null => aborting");
+			return;
+		}
+
+		enableCamera(videoPreviewWindow != null);
+		mLinphoneCore.setPreviewWindow(videoPreviewWindow);
+	}
+
+	private void setFrontCameraAsDefault()
+	{
+		int cameraId = 0;
+		for (final AndroidCameraConfiguration.AndroidCamera androidCamera : AndroidCameraConfiguration.retrieveCameras()) {
+			if (androidCamera.frontFacing) {
+				cameraId = androidCamera.id;
+			}
+		}
+
+		mLinphoneCore.setVideoDevice(cameraId);
+	}
+
+	private void enableCamera(final boolean enable)
+	{
+		Lg.i("enableCamera: ", enable);
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to enable camera for");
+			return;
+		}
+
+		currentCall.enableCamera(enable);
+
+		if (enable) {
+			setFrontCameraAsDefault();
+		}
+
+		mLinphoneCore.updateCall(currentCall, null);
+	}
+
+	public void toggleCamera()
+	{
+		Lg.i("toggleCamera");
+
+		final AndroidCameraConfiguration.AndroidCamera[] cameras = AndroidCameraConfiguration.retrieveCameras();
+		if (cameras.length < 1) {
+			Lg.i("not enough cameras to toggle through");
+			return;
+		}
+
+		final LinphoneCall currentCall = getCurrentCall();
+		if (currentCall == null) {
+			Lg.w("no current call to toggle camera for");
+			return;
+		}
+
+		final int currentCameraId = mLinphoneCore.getVideoDevice();
+
+		for (int i = 0; i < cameras.length; i++) {
+			if (cameras[i].id == currentCameraId) {
+				final int newCameraId = i + 1 < cameras.length ? cameras[i + 1].id : cameras[0].id;
```

Comment:
This looks like a longer version of `final int newCameraId = cameras[(i + 1) % cameras.length]`.

Reply:
You are right and actually I already fixed this in the devel-video branch:
e3177b72d90b69897d4771cb37f32a05b4e8421a